pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -630,6 +630,10 @@ type ValueFrom struct {
 	// JQFilter expression against the resource object in resource templates
 	JQFilter string `json:"jqFilter,omitempty" protobuf:"bytes,3,opt,name=jqFilter"`
 
+	// Selector (https://github.com/antonmedv/expr) that is evaluated against the event to get the value of the parameter. E.g. `payload.message`
+	// +kubebuilder:validation:MinLength=4
+	Expression string `json:"expression,omitempty" protobuf:"bytes,7,opt,name=expression"`
```

Comment:
Why do we need expression in ValueFrom?  As I understand, this field exists for the purposes of parsing an event payload. But we had agreed that we need to decouple the interpretation of events from the workflow/workflow template. The whole purpose of introducing WorkflowEventBinding was to allow that relationship to be decoupled through an intermediary CRD.